src/foam/demos/olympics/Controller.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.demos.olympics',
+  name: 'Controller',
+  extends: 'foam.u2.search.FilterController',
+  requires: [
+    'foam.dao.EasyDAO',
+    'foam.dao.MDAO',
+    'foam.dao.PromisedDAO',
+    'foam.demos.olympics.Medal',
+    'foam.graphics.Canvas',
+    'foam.graphics.ScrollCView',
+    'foam.net.XHRHTTPRequest',
+    'foam.u2.TableView'
+  ],
+
+  constants: {
+    CSS_NAME: 'foam-u2-search-FilterController'
+  },
+
+  properties: [
+    {
+      name: 'data',
+      factory: function() {
+        var dao = this.EasyDAO.create({
+          of: this.Medal,
+          seqNo: true,
+          autoIndex: true,
+          daoType: this.MDAO
+        });
+        var xhr = this.XHRHTTPRequest.create({
+          responseType: 'json',
+          method: 'GET'
+        });
+        xhr.fromUrl('https://raw.githubusercontent.com/foam-framework/foam/' +
+                    'master/js/foam/demos/olympics/MedalData.json');
+        var self = this;
+        return this.PromisedDAO.create({
```

Comment:
Do we need to bother with this?  Does it hurt if we just copy data into the dao as it arrives?